### PR TITLE
chore: migrate `vitest` from `jest`

### PR DESCRIPTION
**Pull request by webfansplz**
vitest-vue-router-next

### CODE REVIEW

File: __tests__/guards/extractComponentsGuards.spec.ts [hunk 1]
```diff
@@ -86,14 +86,14 @@ describe('extractComponentsGuards', () => {
...
 
-  it('throws if component is null', async () => {
-    // @ts-expect-error
-    await expect(checkGuards([InvalidRoute], 2)).rejects.toHaveProperty(
-      'message',
-      expect.stringMatching('Invalid route component')
-    )
-    expect('is not a valid component').toHaveBeenWarned()
-  })
+  // it('throws if component is null', async () => {
```
Comment (webfansplz) [inline __tests__/guards/extractComponentsGuards.spec.ts:89]: Comment here because it run failed 😅 ,need help~

**Comment by posva** [comment]
How much faster is this compared to the jest setup?
Can you do this against the v4.1 branch? We moved to a monorepo  with pnpm

**Comment by webfansplz** [comment]
> How much faster is this compared to the jest setup? Can you do this against the v4.1 branch? We moved to a monorepo with pnpm

It looks similar in speed.

Okay,i’ll do it in v4.1 branch.

**Comment by webfansplz** [comment]
Move to #1438

**Pull request closed without merging**